src/main/kotlin/se/uulm/snowballr/backend/validation/ProjectPaperValidator.kt

```
@@ -0,0 +1,36 @@
+package se.uulm.snowballr.backend.validation
+
+import arrow.core.EitherNel
+import arrow.core.raise.Raise
+import arrow.core.raise.either
+import arrow.core.raise.zipOrAccumulate
+import se.uulm.snowballr.backend.model.ValidationIssue
+import snowballr.ProjectOuterClass.Project.Paper as GrpcProjectPaper
+
+/**
+ * A validator for [GrpcProjectPaper] related requests.
+ */
+object ProjectPaperValidator {
+    fun validateGetRequest(request: GrpcProjectPaper.Get): EitherNel<ValidationIssue, Unit> = either {
+        zipOrAccumulate(
+            { ensureIdValidity("id", request.projectId) },
+            {
+                ensureRelativeProjectPaperIdValidity(request.relativeProjectPaperId)
+            },
+        ) { _, _ -> }
+    }
+
+    /**
+     * Ensures the validity of a given relative project paper ID.
+     *
+     * This function performs the following validations:
+     * - Ensures the ID is non-blank.
+     * - Ensures the ID can be converted to a `Long`.
+     *
+     * @param relativeId The relative project paper ID to validate.
+     */
+    fun Raise<ValidationIssue>.ensureRelativeProjectPaperIdValidity(relativeId: String) {
+        ensureFieldNonBlank("relative_project_paper_id", relativeId)
```

Great! I would have forgotten this check, but actually this is a good solution. 
Anyway, when I looked it up in the specification, I noticed that we don't mention anywhere that you can get an `InvalidArgument` if the ID is incorrect. I think we should add that, what do you think?

---

Yes you are right. Would be a good idea. But one question. Every time we have `Id` as request it could also throw an `InvalidArgument`, but it is never specified. Is this intended for some reason? Asa this is clear, I will create a new issue.

---

Never mind. I just saw, that you've already specified that in [#81](https://github.com/SE-UUlm/snowballr-api/issues/81). I have added your point here to [#81](https://github.com/SE-UUlm/snowballr-api/issues/81).

---

I agree, we should mention this. Lets gather some further problems and then update it in [#81](https://github.com/SE-UUlm/snowballr-api/issues/81) all together